### PR TITLE
Add: Search, Validation: Signup, Login

### PR DESCRIPTION
**Pull request by saurabhchirde**
Added Search Functionality for Notes.

### CODE REVIEW

File: src/Components/UI/Modal/Login.jsx [hunk 1]
```diff
@@ -14,12 +14,20 @@ const Login = () => {
     data: loginInput,
   };
 
+  const emailValidate =
+    /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+
   const onLoginClickFormHandler = () => {
```
Comment (m26unkwn) [inline src/Components/UI/Modal/Login.jsx]: Naming onLoginClickFormHandler can be `onLoginForm`

**Review by m26unkwn — APPROVED** [review]
Wonderful code.

1. Love your UI and The clean way of writing code.

File: src/Components/UI/Navigation/NavBar.jsx [hunk 2]
```diff
@@ -1,45 +1,62 @@
...
 import ButtonIcon from "../Button/ButtonIcon";
 import "./NavBar.css";
 import { useTheme } from "../../../Context/ThemeProvider/ThemeProvider";
+import { useFilter } from "../../../Context";
 
 const NavBar = () => {
   const { darkTheme, setDarkTheme } = useTheme();
+  const { filterDispatch, searchInput, setSearchInput } = useFilter();
+  const location = useLocation();
+  const showSearch = location.pathname.includes("home") ? true : false;
```
Comment (PuneetP16) [inline src/Components/UI/Navigation/NavBar.jsx:14]: Suggestion: you could extract pathname directly from useLocation.

**Review by PuneetP16 — APPROVED** [review]
Looks good to merge, nice work on search functionality.

**Pull request merged**